libs/common/build.gradle

```diff
@@ -43,3 +43,51 @@ tasks.named('forbiddenApisMain').configure {
   // TODO: Need to decide how we want to handle for forbidden signatures with the changes to server
   replaceSignatureFiles 'jdk-signatures'
 }
+
+// Add support for incubator modules on supported Java versions.
+if (BuildParams.runtimeJavaVersion >= JavaVersion.VERSION_20) {
+  sourceSets {
+    java20 {
+      java {
+        srcDirs = ['src/main/java20']
+      }
+    }
+  }
+
+  configurations {
+    java20Implementation.extendsFrom(implementation)
+  }
+
+  dependencies {
+    java20Implementation sourceSets.main.output
+  }
+
+  compileJava20Java {
+    sourceCompatibility = JavaVersion.VERSION_20
```

Comment:
I believe we only need `targetCompatibility` to be set (no language features in used that require source compatibility change)

Reply:
Makes sense. Will update in the next revision.